src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1409,37 +1452,33 @@ void Ekf2::run()
 					lpos.hagl_max = INFINITY;
 				}
 
-				// Get covariances to vehicle odometry
-				float covariances[24];
-				_ekf.get_covariances(covariances);
+				// get pose covariance and copy the URT to the uORB message
+				float *pos_p = &odom.pose_covariance[0];
```

Comment:
Can you avoid doing this with pointers? See Matrix::copyTo().

https://github.com/PX4/Matrix/blob/master/matrix/Matrix.hpp#L85

Reply:
Wea re copying a matrix URT to an array, not the full matrix. How do you suggest that to be done?

Reply:
Solved in e63f8bf